content/doc/book/pipeline/syntax.adoc

```diff
@@ -697,6 +697,13 @@ pipeline {
 
 The `when` directive allows the Pipeline to determine whether the stage should
 be executed depending on the given condition.
+The `when` directive must contain at least one condition.
+If the `when` directive contains more than one condition,
```

Comment:
I disagree. It needs to be mentioned *somewhere* that these two are essentially the same (unless there's further nesting), and if you don't want to mention in the `when` section that it's like `allOf`, it follows that it should be mentioned in `allOf` section that it's like just having multiple statements.